log/aggregator.go

```diff
@@ -0,0 +1,15 @@
+package log
+
+// Aggregator is an interface for pluggable components that collect log messages delivered via
+// some transport mechanism
+type Aggregator interface {
+	// Listen causes an aggregator to begin consuming messages via its underlying transport mechanism
```

Comment:
mind adding docs here that say something along the lines of "this must be a blocking function"